### PR TITLE
Problem (Fix #949): deposit amount fee calculation is not correct

### LINKED ISSUE #949: Problem: "deposit amount" fee calculation is not correct
start tendermint with fee, deposit 500000, but get 500074.

deposited amount in the staked state is larger than what's specified in the command line.

the likely cause is dummysigner estimation

### LINKED ISSUE #1101: Problem: withdraw all coins estimates a wrong fee when extra view keys are entered
steps to reproduce (using client-cli):
1. configure network with non-zero fee -- let one address (A) as unbonded
2. in a different wallet, create transfer address + view key (B)
3. try withdraw all coins from A and provide B details

what happens: validation fails with
> Error: Tendermint RPC error: verification failed: transaction input output coin (plus fee) sums don't match

what should happen -- transaction goes through

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- Fix DummySigner for deposit transaction
- Fix mock_enc_dec payload padding
- Test withdraw with viewkeys in the multinode join test (Fix #1101  )

### CODE REVIEW

File: client-core/src/signer/dummy_signer.rs [hunk 1]
```diff
@@ -25,7 +25,7 @@ impl DummySigner {
         let unit = 16_usize;
         let plain_payload_len = plain_txaux.encode().len();
         // PKCS7 padding -- if the len is multiple of the block length, it'll still be padded
-        vec![0; plain_payload_len + unit - plain_payload_len % unit]
+        vec![0; plain_payload_len + unit - plain_payload_len % unit + 5]
     }
```
Comment (tomtau) [inline client-core/src/signer/dummy_signer.rs]: rfc8452 doesn't seem to have this padding -- seems the length should be `plain_payload_len + tag_len(16)`

Comment (tomtau) [reply client-core/src/signer/dummy_signer.rs]: as Aad is txid, fixed to 32

Comment (yihuang) [reply client-core/src/signer/dummy_signer.rs]: > as Aad is txid, fixed to 32

Yeah, `plain_len + 32` passes the test too, = ="

**Comment by yihuang** [comment]
> also there's a fee computation function in c bindings

`cro_estimate_fee`? that one needs bigger change because it only takes a `tx_payload_size` as an argument, which I think it's the size of the tx before obfuscation, that's not enough to compute the fee.

**Comment by yihuang** [comment]
#1101 is also solved, it's verified in the multinode join test, because normal integration tests will withdraw all at the beginning.

**Comment by tomtau** [comment]
bors try

**Pull request merged**